File: Server/src/main/java/fr/intech/roomies/model/finance/Spending.java
package fr.intech.roomies.model.finance;

import javax.persistence.*;
import java.util.Date;

@Entity
public class Spending {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private int spendingId;

    @Column(nullable = false)
    private Date date;

    @Column(nullable = false)
    private int amount;

    private String description;

    public Spending(){}

    public int getSpendingId() {
        return spendingId;
    }

    public void setSpendingId(int spendingId) {
        this.spendingId = spendingId;
    }

    public Date getDate() {
        return date;
    }

    public void setDate(Date date) {
        this.date = date;
    }

    public int getAmount() {
        return amount;
    }

    public void setAmount(int amount) {
        this.amount = amount;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }
}

File: Roomies2.0/src/Roomies2.DAL/Model/Finance/Spending.cs
#region

using System;

#endregion

namespace Roomies2.DAL.Model.Finance
{
    public class Spending
    {
        public Spending(int spendingId = default, DateTime dateTime = default, int amount = default,
            string description = null)
        {
            SpendingId = spendingId;
            DateTime = dateTime;
            Amount = amount;
            Description = description ?? throw new ArgumentNullException(nameof(description));
        }

        public int SpendingId { get; set; }
        public DateTime DateTime { get; set; }
        public int Amount { get; set; }
        public string Description { get; set; }
    }
}